runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -133,160 +138,41 @@ public void mergeSegments(@Nonnull String endpointToRestoreRedundancy, int retry
-
-        return Stream.of(getLayoutServersClusterHealth(layout, peerResponsiveNodes),
-                getSequencerServersClusterHealth(layout, peerResponsiveNodes),
-                getLogUnitServersClusterHealth(layout, peerResponsiveNodes))
-                // Gets cluster status from the layout, sequencer and log unit clusters.
-                // The status is then aggregated by the max of the 3 statuses acquired.
-                .max(Comparator.comparingInt(ClusterStatus::getHealthValue))
-                .orElse(ClusterStatus.UNAVAILABLE);
-    }
-
-    /**
-     * Returns a LogUnit Server's status in the layout. It is marked as:
-     * UP if it is present in all segments or none of the segments and not in the unresponsive list,
-     * NOTE: A node is UP if its not in any of the segments as it might not be a LogUnit component
-     * but has only the Layout or the Sequencer (or both) component(s) active.
-     * DB_SYNCING if it is present in some but not all or none of the segments,
-     * DOWN if it is present in the unresponsive servers list.
-     *
-     * @param layout Layout to check.
-     * @param server LogUnit Server endpoint.
-     * @return NodeState with respect to the layout specified.
-     */
-    private NodeStatus getLogUnitNodeStatusInLayout(Layout layout, String server) {
-        if (layout.getUnresponsiveServers().contains(server)) {
-            return NodeStatus.DOWN;
-        }
-        final int segmentsCount = layout.getSegments().size();
-        int nodeInSegments = 0;
-        for (LayoutSegment layoutSegment : layout.getSegments()) {
-            if (layoutSegment.getAllLogServers().contains(server)) {
-                nodeInSegments++;
-            }
-        }
-        return nodeInSegments == segmentsCount || nodeInSegments == 0
-                ? NodeStatus.UP : NodeStatus.DB_SYNCING;
-    }
-
     /**
      * Get the Cluster Status.
-     *
+     * <p>
      * This is reported as follows:
      * - (1) The status of the cluster itself (regardless of clients connectivity) as reflected in the
-     *   layout. This information is presented along each node's status (up, down, db_sync).
-     *
-     *   It is important to note that as the cluster state is obtained from the layout,
-     *   when quorum is not available (majority of nodes) there are lower guarantees on the
-     *   reliability of this state.
-     *   For example, in the absence of quorum the system might be in an unstable state which
-     *   cannot converge due to lack of consensus. This is reflected in the
-     *   cluster status report as clusterStatusReliability.
-     *
+     * layout. This information is presented along each node's status (up, down, db_sync).
+     * <p>
+     * It is important to note that as the cluster state is obtained from the layout,
+     * when quorum is not available (majority of nodes) there are lower guarantees on the
+     * reliability of this state.
```

Comment:
shouldnt the cluster state always be 'unstable' in the absence of a quorum?

Reply:
As I know, on the client-side it's not possible to know exactly what the cluster state in case of a failure, but if the client can't connect to a server in the cluster it doesn't mean the cluster in the DEGRADED state.
@annym can explain better than me.

Reply:
exactly, the fact that a client cannot connect to all 3 servers, does not necessarily imply that the Cluster is UNSTABLE, probably it's all up and running, it's just this client who cannot connect to it. That is why reading the cluster status should consider reliability and connection status, to get an overall look.